src/ansi-c/c_nondet_symbol_factory.cpp

```diff
@@ -216,6 +233,25 @@ void symbol_factoryt::gen_nondet_init(
   }
 }
 
+void symbol_factoryt::gen_nondet_array_init(
+  code_blockt &assignments,
+  const exprt &expr,
+  std::size_t depth,
+  const recursion_sett &recursion_set)
+{
+  auto const &array_type = to_array_type(expr.type());
+  auto const array_size = numeric_cast_v<mp_integer>(array_type.size());
```

Comment:
Use `numeric_cast<std::size_t>` and equally use `std::size_t` in the `for`-loop below. If that array is larger than the maximum value representable on the machine we are running on then the loop below wouldn't succeed either...